csharp/Microsoft.Azure.Databricks.Client/PipelinesApiClient.cs

```diff
@@ -0,0 +1,223 @@
+        string untilUpdateId = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUriSb = new StringBuilder($"{ApiVersion}/pipelines/{pipelineId}/updates");
+            
+        if (pageToken != null || maxResults != null || untilUpdateId != null)
+        {
+            requestUriSb.Append('?');
+        }
+
+        if (pageToken != null)
+        {
+            requestUriSb.Append($"&{pageToken}");
+        }
+
+        if (maxResults != null)
+        {
+            requestUriSb.Append($"&{maxResults}");
+        }
+
+        if (untilUpdateId != null)
+        {
+            requestUriSb.Append($"&{untilUpdateId}");
+        }
+
+        var requestUri = requestUriSb.ToString();
+        return await HttpGet<PipelineUpdatesList>(this.HttpClient, requestUri, cancellationToken).ConfigureAwait(false);
+    }
+
+    public async Task<string> QueueUpdate(
+        string pipelineId,
+        bool fullRefresh = false,
+        PipelineUpdateCause cause = PipelineUpdateCause.API_CALL,
+        string refreshSelection = default,
+        string fullRefrehSelection = default,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUri = $"{ApiVersion}/pipelines/{pipelineId}/updates";
+
+        var requestDict = new Dictionary<string, string>()
+        {
+            { "full_refresh", fullRefresh.ToString() },
+            { "cause", cause.ToString() }
+        };
+
+        if (refreshSelection != null)
+        {
+            requestDict["refresh_selection"] = refreshSelection;
+        }
+
+        if (fullRefrehSelection != null)
+        {
+            requestDict["full_refresh_selection"] = fullRefrehSelection;
+        }
+
+        var request = JsonSerializer.SerializeToNode(requestDict, Options).AsObject();
+        var response = await HttpPost<JsonObject, JsonObject>(this.HttpClient, requestUri, request, cancellationToken).ConfigureAwait(false);
+
+        return response["update_id"].GetValue<string>();
+    }
+
+    public async Task<PipelineEventsList> ListEvents(
+        string pipelineId,
+        int? maxResults = null,
+        string orderBy = null,
+        string filter = null,
+        string pageToken = null,
+        CancellationToken cancellationToken = default)
+    {
+        var requestUriSb = new StringBuilder($"{ApiVersion}/pipelines/{pipelineId}/events");
+
+        if (maxResults != null || orderBy != null || filter != null || pageToken != null)
+        {
+            requestUriSb.Append('?');
+        }
+
+        if (maxResults != null)
+        {
+            requestUriSb.Append($"&{maxResults}");
+        }
+
+        if (orderBy != null)
+        {
+            requestUriSb.Append($"&{orderBy}");
+        }
+
+        if (filter != null)
+        {
+            requestUriSb.Append($"&{filter}");
+        }
+        
+        if (pageToken != null)
+        {
+            requestUriSb.Append($"&{pageToken}");
+        }

```

Comment:
Query string format is also wrong here.

Reply:
Very stupid mistake. I'm wondering - there is nothing about default max_results value in doc. Do you have any idea what could be default value to simplify logic a little?

Reply:
@skarpecki , the public document didn't specifically call out the default value, but 25 should be a fair value to use since it is the default value for the same parameter in the pipelines/list API call.